src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -196,7 +197,9 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
                     using (var recoveryPager = _env.Options.CreateTemporaryBufferPager(journalRecoveryName, initialSize))
                     using (var pager = _env.Options.OpenJournalPager(journalNumber, logInfo))
                     {
-                        RecoverCurrentJournalSize(pager);
+                        var isMoreThanMaxFileSize = RecoverCurrentJournalSizeAndCheckIfLargerThanMaxLogFileSize(pager);
+                        if (journalNumber == logInfo.CurrentJournal)
```

Comment:
At this point we're sure that `CurrentJournal` is the last one, right? Maybe add some debug assertion which will verify that.

Reply:
yes we are sure because of the for clause
https://github.com/ravendb/ravendb/blob/7096a085e174f62f001d6cfc5df95f68186e1392/src/Voron/Impl/Journal/WriteAheadJournal.cs#L190

Reply:
okay